compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java

```diff
@@ -188,64 +184,63 @@ public static RemotePackageRepository from(Environment environment, Path cacheDi
 
     private List<ImportModuleResponse> mergeNameResolution(Collection<ImportModuleResponse> filesystem,
                                                            Collection<ImportModuleResponse> remote) {
-        List<ImportModuleResponse> all = new ArrayList<>();
-        // We assume file system responses will have all module requests
-        for (ImportModuleResponse fileResponse : filesystem) {
-            if (fileResponse.resolutionStatus() == ResolutionResponse.ResolutionStatus.RESOLVED) {
-                all.add(fileResponse);
-            } else {
-                // If remote has resolutions of unresolved we will substitute
-                Optional<ImportModuleResponse> remoteResponse = remote.stream()
-                        .filter(r -> r.importModuleRequest().equals(fileResponse.importModuleRequest())).findFirst();
-                if (remoteResponse.isPresent()) {
-                    all.add(remoteResponse.get());
-                }
-            }
-        }
-        return all;
+        return new ArrayList<>(
+            Stream.of(filesystem, remote)
+                .flatMap(Collection::stream).collect(Collectors.toMap(
+                    ImportModuleResponse::importModuleRequest, Function.identity(),
+                    (ImportModuleResponse x, ImportModuleResponse y) -> {
+                        if (y.resolutionStatus().equals(ResolutionResponse.ResolutionStatus.UNRESOLVED)) {
+                            return x;
+                        } else if (x.resolutionStatus().equals(ResolutionResponse.ResolutionStatus.UNRESOLVED)) {
+                            return y;
+                        } else if (x.packageDescriptor().version().compareTo(
```

Comment:
Semantic version comparison will not work for initial versions since they are considered INCOMPATIBLE with one another. Let's use the `ProjectUtis.getLatest` util for this. 
FYR: https://github.com/ballerina-platform/ballerina-lang/blob/849496f3e17fb0ec8bbc1bce9832e8b5972c4c46/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java#L309

Reply:
Added